pkg/operator/apiserver/audit/audit_policies_test.go

```diff
@@ -83,14 +85,58 @@ func TestGetAuditPolicy(t *testing.T) {
 			},
 			goldenFile: "oauth.yaml",
 		},
-		// TODO: add test with multiple customRules
+		{
+			name: "multipleCustomRules",
+			config: configv1.Audit{
+				Profile: "None",
+				CustomRules: []configv1.AuditCustomRule{
+					{
+						Group:   "system:authenticated:oauth",
+						Profile: "WriteRequestBodies",
+					},
+					{
+						Group:   "system:authenticated",
+						Profile: "AllRequestBodies",
+					},
+				},
+			},
+			goldenFile: "multipleCr.yaml",
+		},
+		{
+			name: "unknownProfile",
+			config: configv1.Audit{
+				Profile: "InvalidString",
+			},
+			errContains: "unknown audit profile \"InvalidString\"",
+		},
+		{
+			name: "unknownCustomRulesProfile",
+			config: configv1.Audit{
+				Profile: "None",
+				CustomRules: []configv1.AuditCustomRule{
+					{
+						Group:   "InvalidGroup",
+						Profile: "InvalidProfile",
+					},
+				},
+			},
+			errContains: "unknown audit profile \"InvalidProfile\" in customRules for group \"InvalidGroup\"",
+		},
 	}
 	for _, scenario := range scenarios {
 		t.Run(scenario.name, func(t *testing.T) {
 			// act
 			policy, err := GetAuditPolicy(scenario.config)
-			if err != nil {
-				t.Fatal(err)
+			if len(scenario.errContains) == 0 && err != nil {
+				t.Fatalf("Expected no error yet received error: %v", err)
+			}
+			if len(scenario.errContains) > 0 {
+				if err == nil {
+					t.Fatalf("Expected error message: %q, but got: %q", scenario.errContains, err.Error())
+				}
+				if strings.Contains(err.Error(), scenario.errContains+",") {
```

Comment:
why comma?

Reply:
It needs it
```=== RUN   TestGetAuditPolicy/unknownProfile
    audit_policies_test.go:142: Expected error message: "unknown audit profile \"InvalidString\"", but got: "unknown audit profile \"InvalidString\""
=== RUN   TestGetAuditPolicy/unknownCustomRulesProfile
    audit_policies_test.go:142: Expected error message: "unknown audit profile \"InvalidProfile\" in customRules for group \"InvalidGroup\"", but got: "unknown audit profile \"InvalidProfile\" in customRules for group \"InvalidGroup\""
--- FAIL: TestGetAuditPolicy (0.00s)```

Reply:
Read carefully, it is there. Also mentioned in Slack

Reply:
it does not need the comma, the issue here is you are failing the test if the expected error message is contained in the actual error message.
```
if strings.Contains(err.Error(), scenario.errContains) {
// fail the test
}
```

Reply:
Should be resolved with last commit 👍